File: base/src/main/groovy/com/knight/plugin/doublecheck/DoubleCheckWeaver.java
package com.knight.plugin.doublecheck;

import com.knight.plugin.doublecheck.java.DoubleClickCheckModifyClassAdapter;
import com.knight.plugin.doublecheck.java.WeavedClass;
import com.knight.plugin.doublecheck.transform.BaseWeaver;

import org.objectweb.asm.ClassVisitor;
import org.objectweb.asm.ClassWriter;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;

public class DoubleCheckWeaver extends BaseWeaver {

    private ArrayList<WeavedClass> weavedClasses;
    private DoubleClickCheckModifyClassAdapter adapter;

    public DoubleCheckWeaver(ArrayList<WeavedClass> weavedClasses) {
        this.weavedClasses = weavedClasses;
    }

    @Override
    protected ClassVisitor wrapClassWriter(ClassWriter classWriter) {
        adapter = new DoubleClickCheckModifyClassAdapter(classWriter);
        return adapter;
    }

    @Override
    public byte[] weaveSingleClassToByteArray(InputStream inputStream) throws IOException {
        byte[] bytes = super.weaveSingleClassToByteArray(inputStream);
        weavedClasses.add(adapter.getWeavedClass());
        return bytes;
    }
}

File: transform/src/main/java/com/knight/transform/DoubleCheckWeaver.kt
package com.knight.transform

import com.knight.transform.asm.BaseWeaver
import com.knight.transform.java.DoubleClickCheckModifyClassAdapter
import com.knight.transform.java.outmap.WeavedClass
import org.objectweb.asm.ClassVisitor
import org.objectweb.asm.ClassWriter

class DoubleCheckWeaver(var weaverClasses: ArrayList<WeavedClass>) : BaseWeaver() {


    override fun wrapClassWriter(classWriter: ClassWriter): ClassVisitor {
        return DoubleClickCheckModifyClassAdapter(classWriter)
    }

    override fun getInfomation(classVisitor: ClassVisitor) {
//        println("test::::: " + (classVisitor as DoubleClickCheckModifyClassAdapter).weavedClass.toString())
        weaverClasses.add((classVisitor as DoubleClickCheckModifyClassAdapter).weavedClass)
    }

}